ImageLoader/JPEGImageOperations/JPEGImageOperations.swift

```diff
@@ -0,0 +1,151 @@
+    case ARGB0888 = 8 // TJPF_XRGB
+    case ABGR0888 = 9 // TJPF_XBGR
+    case YUV400 = 10 // TJPF_GREY
+
+    var channelCount: Int32 {
+        switch self {
+        case .RGB888:
+            return 3
+        case .BGR888:
+            return 3
+        case .RGBA8888:
+            return 4
+        case .BGRA8888:
+            return 4
+        case .ARGB8888:
+            return 4
+        case .ABGR8888:
+            return 4
+        case .RGBA8880:
+            return 3
+        case .BGRA8880:
+            return 3
+        case .ARGB0888:
+            return 3
+        case .ABGR0888:
+            return 3
+        case .YUV400:
+            return 1
+        }
+    }
+}
+
+public class ImageData {
+    let height: Int32
+    let width: Int32
+    let buffer: UnsafeMutablePointer<UInt8>
+    let formatProperties: pixelFormat
+
+    init(height: Int32, width: Int32, imageFormat: pixelFormat) {
+        self.height = height
+        self.width = width
+        formatProperties = imageFormat
+        buffer = tjAlloc(imageFormat.channelCount * width * height)
+    }
+
+    deinit {
+        tjFree(self.buffer)
+    }
+}
+
+func loadJPEG(atPath path: String, imageFormat: pixelFormat) throws -> ImageData {
+    var width: Int32 = 0
+    var height: Int32 = 0
+
+    guard FileManager.default.fileExists(atPath: path) else {
+        throw ("File does not exist at \(path).")
+    }
+
+    let data: Data = FileManager.default.contents(atPath: path)!
+    let jpegSize: UInt = UInt(data.count)
+
+    return data.withUnsafeBytes {
+        let finPointer: UnsafeMutablePointer<UInt8> = UnsafeMutablePointer<UInt8>(mutating: $0.baseAddress!.assumingMemoryBound(to: UInt8.self))
+
+        let decompressor = tjInitDecompress()
+        defer { tjDestroy(decompressor) }
+
+        /* Initializes `width` and `height` variables */
+        tjDecompressHeader(decompressor, finPointer, jpegSize, &width, &height)
+
+        let imageData = ImageData(height: height, width: width, imageFormat: imageFormat)
+
+        /* Decompresses the JPEG Image from `data` into `buffer` buffer
+         - Decompresses `finPointer` which has image data
+         - uses `jpegSize` as size of image in bytes
+         - Image gets decompressed into `buffer` buffer
+         - `width` = width of Image
+         - pitch = 0
+         - `height` = height of image
+         - pixelFormat = imageFormat.rawValue which denotes Pixel Format to which image is being decompressed.
+         - flags = 0
+         */
+        tjDecompress2(decompressor, finPointer, UInt(jpegSize), imageData.buffer, imageData.width, 0, imageData.height, imageFormat.rawValue, 0)
+
+        return imageData
+    }
+}
+
+func saveJPEG(atPath path: String, image: ImageData) throws {
+    do {
+        try FileManager.default.removeItem(atPath: path)
+    } catch {
+        // File not present
+    }
+
+    var jpegBuf: UnsafeMutablePointer<UInt8>?
+    defer { tjFree(jpegBuf) }
+
+    let outQual: Int32 = 95
```

Comment:
```suggestion
    let outQual: CInt = 95
```

This seems like an inout parameter, what do you think of renaming it to `quality`?

Reply:
I renamed it.